src/app/query/global-query/global-query.component.ts

```diff
@@ -0,0 +1,159 @@
+import { DatasetSubscriptionsService } from "src/app/dataset-view/dataset.subscriptions.service";
+import ProjectLinks from "src/app/project-links";
+
+@Component({
+    selector: "app-global-query",
+    templateUrl: "./global-query.component.html",
+    styleUrls: ["./global-query.component.scss"],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class GlobalQueryComponent extends BaseComponent implements OnInit {
+    public sqlRequestCode = "";
+    public sqlLoading = false;
+    public searchResult: GlobalQuerySearchItem[] = [];
+
+    //Search
+    public inputDatasets = new Set<string>();
+    public searchDataset = "";
+    private readonly delayTime: number = AppValues.SHORT_DELAY_MS;
+
+    private datasetService = inject(DatasetService);
+    private cdr = inject(ChangeDetectorRef);
+    private appSearchAPI = inject(SearchApi);
+    private datasetSubsService = inject(DatasetSubscriptionsService);
+
+    ngOnInit(): void {
+        this.initSqlQueryFromUrl();
+        this.datasetSubsService.emitInvolvedSqlDatasetsId([]);
+        this.datasetSubsService.involvedSqlDatasetsIdChanges
+            .pipe(
+                switchMap((ids: string[]) => {
+                    const schemaResponses = ids.map((id: string) => this.datasetService.requestDatasetSchema(id));
+                    return this.processSchemaResponses(schemaResponses);
+                }),
+                takeUntilDestroyed(this.destroyRef),
+            )
+            .subscribe((data: GlobalQuerySearchItem[]) => {
+                this.searchResult = data;
+                this.cdr.detectChanges();
+            });
+    }
+
+    private initSqlQueryFromUrl(): void {
+        const sqlQueryFromUrl = this.activatedRoute.snapshot.queryParamMap.get(ProjectLinks.URL_QUERY_PARAM_SQL_QUERY);
+        if (sqlQueryFromUrl) {
+            this.sqlRequestCode = sqlQueryFromUrl;
+        }
+    }
+
+    private processSchemaResponses(
+        schemaResponses: Observable<GetDatasetSchemaQuery>[],
+    ): Observable<GlobalQuerySearchItem[]> {
+        return combineLatest(schemaResponses).pipe(
+            map((datasets: GetDatasetSchemaQuery[]) => {
+                const result = datasets.map((item: GetDatasetSchemaQuery) => {
+                    const datasetAlias = item.datasets.byId?.alias;
+                    const schema = parseCurrentSchema(item.datasets.byId?.metadata.currentSchema);
+                    return { datasetAlias, schema } as GlobalQuerySearchItem;
+                });
+                return result;
+            }),
+        );
+    }
+
+    public deleteDataset(datasetAlias: string): void {
+        this.searchResult = this.searchResult.filter(
+            (item: GlobalQuerySearchItem) => item.datasetAlias !== datasetAlias,
+        );
+        this.inputDatasets.forEach((item) => {
+            if (item.includes(datasetAlias)) {
+                this.inputDatasets.delete(item);
+            }
+        });
+    }
```

Comment:
You are not deleting the dataset. You are just removing it from loaded inputs.

Reply:
Fixed